embeddings/embedding/flair_loader.py

```diff
@@ -0,0 +1,70 @@
+from abc import abstractmethod
+from pathlib import Path
+from typing import Any, Type, Union
+
+from embeddings.data.io import T_path
+from embeddings.embedding.auto_flair import (
+    AutoFlairDocumentEmbedding,
+    AutoFlairDocumentPoolEmbedding,
+    AutoFlairWordEmbedding,
+    DocumentEmbedding,
+)
+from embeddings.embedding.flair_embedding import FlairDocumentPoolEmbedding, FlairEmbedding
+
+
+class FlairEmbeddingLoader:
+    def __init__(self, embedding_name: T_path, model_type_reference: str):
+        self.embedding_name = embedding_name
+        self.model_type_reference = model_type_reference
+
+    @abstractmethod
+    def get_embedding(self) -> FlairEmbedding:
+        """Loads an embedding model from hugging face hub or from file that is stored locally,
+        if the model is compatible with Transformers or the current library
+        """
+        pass
```

Comment:
It is possible to use as an argument the object of the class that returns AutoFlair embedding but then the code would be less readable. It might be considered during refactor of the entire package.

Reply:
I think it should be done before merging to make refactoring easier.

Reply:
I think it should be done before merging to make refactoring easier.